src/osp/Active/SysVehicle.h

```diff
@@ -111,17 +112,77 @@ class SysVehicle : public IDynamicSystem, public IActivator
             universe::Satellite areaSat, universe::Satellite tgtSat,
             ActiveEnt tgtEnt);
 
-    /**
-     * Create a Physical Part from a PrototypePart and put it in the world
-     * @param part the part to instantiate
-     * @param rootParent Entity to put part into
-     * @return Pointer to object created
-     */
-    ActiveEnt part_instantiate(PrototypePart& part, ActiveEnt rootParent);
 
     // Handle deleted parts and separations
     void update_vehicle_modification(ActiveScene& rScene);
 
+private:
+
+    // Stores the association between a PrototypeObj entity and the indices of
+    // its owned machines in the PrototypePart array
+    struct MachineDef
+    {
+        ActiveEnt m_machineOwner;
+        std::vector<unsigned> const& m_machineIndices;
+
+        MachineDef(ActiveEnt owner, std::vector<unsigned> const& indexArray)
+            : m_machineOwner(owner)
+            , m_machineIndices(indexArray)
+        {}
+    };
+
+    /**
+     * Add machines to the specified entity
+     *
+     * Receives the master arrays of prototype and blueprint machines from a
+     * prototype/blueprint part, as well as an array of indices that describe
+     * which of the machines from the master list are to be instantiated for
+     * the specified entity.
+     * @param partEnt [in] The part entity which owns the ACompMachines component
+     * @param entity [in] The target entity to receive the machines
+     * @param protoMachines [in] The master array of prototype machines
+     * @param blueprintMachines [in] The master array of machine configs
+     * @param machineIndices [in] The indices of the machines to add to entity
+     */
+    void add_machines_to_object(ActiveEnt partEnt, ActiveEnt entity,
+        std::vector<PrototypeMachine> const& protoMachines,
+        std::vector<BlueprintMachine> const& blueprintMachines,
+        std::vector<unsigned> const& machineIndices);
```

Comment:
if some third party library is using "unsigned" as an index into itself, typedef that to a named type. E.g. LibOtherIndex_t or similar.